### PR TITLE
+eigen.tuxfamily.org

### PR DESCRIPTION
**Pull request by sanchitram1**
One of the homebrew packages, see https://github.com/teaxyz/pantry.extra/issues/372

### CODE REVIEW

File: projects/eigen.tuxfamily.org/package.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  url: https://gitlab.com/libeigen/eigen/-/archive/{{version}}/eigen-{{version}}.tar.gz
+  strip-components: 1
+
+versions:
+  - 3.4.0
+
+build:
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
```
Comment (mxcl) [inline projects/eigen.tuxfamily.org/package.yml:12]: `cmake.org: ^3`

**Comment by sanchitram1** [comment]
- Replaced the `mkdir...` and `cd...` steps with a `working-directory: eigen-build` key in yaml
- Added `cmake.org` as a build & test dependency
- Since eigen is a header-only library, no need for the `provides` tag (no binaries)
- Since it's header only, you have to specify eigen3 in the include path. This is stored in `cxxflags`
- Replaced diff with a regular string comparison.

**Comment by sanchitram1** [comment]
Closes #372

File: projects/eigen.tuxfamily.org/package.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  - 3.4.0
+
+build:
+  working-directory: ./eigen-build 
+  dependencies:
+    tea.xyz/gx/cc: c99
+    tea.xyz/gx/make: '*'
+    cmake.org: '^3'
+  script: |
+    pwd
```
Comment (mxcl) [inline projects/eigen.tuxfamily.org/package.yml]: innocuous but shouldn't be here

File: projects/eigen.tuxfamily.org/package.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    cmake .. -DCMAKE_INSTALL_PREFIX={{prefix}}
```
Comment (mxcl) [inline projects/eigen.tuxfamily.org/package.yml:15]: Should add `-DCMAKE_BUILD_TYPE=Release`

**Comment by mxcl** [comment]
Note I'm doing a newish thing here where if the includes are eg. `include/NAMESPACE/foo` we make NAMESPACE a symlink to `include` and mv foo to `include/foo`.

This is for robustness, it will fix a lot of potential failures in builds for both us and our users.

We can do this because we don’t install things.

**Pull request merged**